tests/accelerators/test_accelerator_connector.py

```diff
@@ -639,3 +639,50 @@ def test_accelerator_cpu_with_multiple_gpus():
 
     assert trainer._device_type == "cpu"
     assert isinstance(trainer.accelerator, CPUAccelerator)
+
+
+@pytest.mark.parametrize(["devices", "plugin"], [(1, SingleDevicePlugin), (5, DDPSpawnPlugin)])
+def test_accelerator_cpu_with_devices(devices, plugin):
+
+    trainer = Trainer(accelerator="cpu", devices=devices)
+
+    assert trainer.num_processes == devices
+    assert isinstance(trainer.training_type_plugin, plugin)
+    assert isinstance(trainer.accelerator, CPUAccelerator)
+
+
+def test_accelerator_cpu_with_num_processes_priority():
+    """ Test for checking num_processes takes priority over devices. """
+
+    num_processes = 5
+    trainer = Trainer(accelerator="cpu", devices=8, num_processes=num_processes)
```

Comment:
I would prefer to raise a warning or exception instead.

Reply:
We are planning to deprecate `num_processes` soon, and `devices` would be the go-to flag and would have `1` default value.

Reply:
I think we should still raise a DeprecationWarning in that case @kaushikb11

Reply:
@justusschock We would be adding a deprecation warning in the follow-up PR.